ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx

```diff
@@ -8,10 +8,12 @@ import {Loading} from '../../../shared/components/loading';
 import {services} from '../../../shared/services';
 import {CronWorkflowSummaryPanel} from '../cron-workflow-summary-panel';
 
+const jsonMergePatch = require('json-merge-patch');
 require('../../../workflows/components/workflow-details/workflow-details.scss');
 
 interface State {
     cronWorkflow?: CronWorkflow;
+    editing: boolean;
```

Comment:
I'm not too sure we need this... it would make sense to me if I was editing the YAML and _without saving my edits_ pressed Suspend/Resume that my changes would be lost.

Reply:
This makes sense too, but I think this is safer. The ideal design would be having an additional `cancel` button besides the `submit` button while still disabling the suspend button during editing. I think this flow is more natural and ensures safety simultaneously.

Reply:
I agree that it's safer, but I can see some potential for edge case bugs. For example, in this implementation this flag is only set to false if the underlying object is changed. Does this flag work correctly if a user clicks edit then save without making any changes?

I would rather move forward without this change for now, and add it later if necessary. Let me know if you feel strongly otherwise.

Reply:
You're right. To solve this, we should not set `editing: false` in `onChange` but do this in another function and pass it in as a separate props. This should be called after `.then(updated => props.onChange(updated))`.

I don't feel this is troublesome or will severely affect other things but can improve UX a little bit, so to me it's still beneficial. If you still have concerns, we can skip this now.

Reply:
Let's skip this for now please. You're welcome to open a separate PR with this change, though. If you do, please make sure that it's consistent across different resource kinds (`Workflow`, `WorkflowTemplate`, etc.)